core/src/main/java/org/apache/carbondata/core/datastore/page/UnsafeFixLengthColumnPage.java

```diff
@@ -369,7 +367,7 @@ public BigDecimal getDecimal(int rowId) {
 
   @Override
   public double[] getDoublePage() {
-    double[] data = new double[getPageSize()];
+    double[] data = new double[getEndLoop()];
```

Comment:
the return values of getPageSize() and getEndLoop seem to be same, when they are diff?

Reply:
during complex type enhancement, to convert value and to get the value, pageSize was changed and new method called getEndLoop added, this was missed for double, so when double datatype is there in complex type data miss match happens, this is handled here. you can refer #2417

Reply:
get it， thank you